src/substrait/common/Decimal.cpp

```diff
@@ -0,0 +1,49 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include "substrait/common/Decimal.h"
+
+namespace io::substrait::common {
+
+Decimal Decimal::fromProto(
+    const ::substrait::proto::Expression_Literal_Decimal& proto) {
+  return {proto.value(), proto.precision(), proto.scale()};
+};
+
+std::string Decimal::toString() {
+  if (value_.empty()) {
```

Comment:
I added an optional validity check (optional because a system may be able to use the unchecked version for increased speed).  Regardless this simple check guards a potential access violation so it's hard for me to remove.